### PR TITLE
Updated Sub layer dialog to display the correct relative path when enabled.

### PR DESCRIPTION
**Pull request by AramAzhari-adsk**

### CODE REVIEW

File: plugin/adsk/scripts/mayaUsd_USDRootFileRelative.py [hunk 1]
```diff
@@ -240,7 +241,12 @@ def updateFilePathPreviewFields(cls, makePathRelative=None):
             # the user is free to enter any extension they want. The layer editor code will then verify
             # (and fix if needed) the file path before saving. We do the same here for preview.
             unresolvedPath = mayaUsdLib.Util.ensureUSDFileExtension(selectedFile) if selectedFile else ''
-            relativePath = mayaUsdLib.Util.getPathRelativeToMayaSceneFile(unresolvedPath) if unresolvedPath else ''
+            """relativePath = mayaUsdLib.Util.getPathRelativeToDirectory(unresolvedPath, cls._parentLayerPath) if unresolvedPath and cls._parentLayerPath != "" else ''"""
```
Comment (seando-adsk) [inline plugin/adsk/scripts/mayaUsd_USDRootFileRelative.py:244]: Was this commented code replaced by below? If so it should be removed.

**Review by seando-adsk — CHANGES_REQUESTED** [review]

**Comment by seando-adsk** [comment]
@pierrebai-adsk Can you have a look at these changes as well? You have done most of the relative path work.

File: plugin/adsk/scripts/mayaUsd_USDRootFileRelative.py (hunk 1 above)
Comment (AramAzhari-adsk) [reply plugin/adsk/scripts/mayaUsd_USDRootFileRelative.py]: removed the comment

**Review by seando-adsk — APPROVED** [review]

File: lib/mayaUsd/python/wrapUtil.cpp [hunk 2]
```diff
@@ -56,14 +56,16 @@ std::string ensureUSDFileExtension(const std::string& fileToCheck)
...
-                  .staticmethod("prettifyName")
-                  .def("getDictionaryFromEncodedOptions", getDictionaryFromEncodedOptions)
-                  .def(
-                      "getPathRelativeToMayaSceneFile",
-                      UsdMayaUtilFileSystem::getPathRelativeToMayaSceneFile)
-                  .def("ensureUSDFileExtension", ensureUSDFileExtension)
-                  .staticmethod("getPathRelativeToMayaSceneFile");
+    scope s
+        = class_<UsdMayaUtilScope>("Util", no_init)
+              .def("IsAuthored", UsdMayaUtil::IsAuthored)
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/python/wrapUtil.cpp:61]: For the life of me, I cannot figure out why clang-format decided all this needed to be reformatted. The rules are so obscure!

File: lib/mayaUsd/utils/utilFileSystem.cpp [hunk 3]
```diff
@@ -174,6 +174,23 @@ std::pair<std::string, bool> UsdMayaUtilFileSystem::makePathRelativeTo(
...
 
+std::string UsdMayaUtilFileSystem::getPathRelativeToDirectory(
+    const std::string& fileName,
+    const std::string& relativeToDir)
+{
+    auto relativePathAndSuccess = makePathRelativeTo(fileName, relativeToDir);
+
+    if (!relativePathAndSuccess.second) {
+        TF_WARN(
+            "File name (%s) cannot be resolved as relative to its parent layer directory (%s), "
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/utils/utilFileSystem.cpp:185]: This function is general, so I would think we should not mention "parent layer" in the error.

**Comment by AramAzhari-adsk** [comment]
@pierrebai-adsk I took notes and will keep a commit to be checked in in the future alongside refactors / relative path work. 
Thanks.

**Pull request merged**